core/vm/environment.libevm.go

```diff
@@ -118,13 +132,17 @@ func (e *environment) callContract(typ CallType, addr common.Address, input []by
 			}
 		case nil:
 		default:
-			return nil, gas, fmt.Errorf("unsupported option %T", o)
+			return nil, fmt.Errorf("unsupported option %T", o)
 		}
 	}
 
 	if in.readOnly && value != nil && !value.IsZero() {
-		return nil, gas, ErrWriteProtection
+		return nil, ErrWriteProtection
```

Comment:
nit same here it + add context

```suggestion
		return nil,fmt.Errorf("%w: interpreter is ready only but value is %s", ErrWriteProtection, value)
```

Reply:
As with https://github.com/ava-labs/libevm/pull/73#discussion_r1888899560